x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java

```diff
@@ -391,21 +399,52 @@ private void startComputeOnRemoteClusters(
                     queryPragmas.exchangeBufferSize(),
                     esqlExecutor,
                     refs.acquire().delegateFailureAndWrap((l, unused) -> {
-                        var remoteSink = exchangeService.newRemoteSink(rootTask, childSessionId, transportService, cluster.connection);
-                        exchangeSource.addRemoteSink(remoteSink, true, queryPragmas.concurrentExchangeClients(), ActionListener.noop());
                         var remotePlan = new RemoteClusterPlan(plan, cluster.concreteIndices, cluster.originalIndices);
                         var clusterRequest = new ClusterComputeRequest(cluster.clusterAlias, childSessionId, configuration, remotePlan);
-                        var clusterListener = ActionListener.runBefore(
-                            computeListener.acquireCompute(cluster.clusterAlias()),
-                            () -> l.onResponse(null)
+
+                        // Create group task for this cluster
+                        CancellableTask groupTask = createGroupTask(rootTask, clusterRequest.getDescription());
+                        CountDown countDown = new CountDown(2);
+                        // The group is done when both the sink and the cluster request are done
+                        Runnable finishGroup = () -> {
+                            if (countDown.countDown()) {
+                                taskManager.unregister(groupTask);
+                                l.onResponse(null);
+                            }
+                        };
+                        // Cancel the group on sink failure
+                        ActionListener<Void> exchangeListener = computeListener.acquireAvoid().delegateResponse((inner, e) -> {
+                            taskManager.cancelTaskAndDescendants(groupTask, "exchange sink failure", true, ActionListener.noop());
+                            if (suppressRemoteFailure) {
+                                computeListener.markAsPartial(clusterAlias, e);
+                                inner.onResponse(null);
+                            } else {
+                                inner.onFailure(e);
+                            }
+                        });
+                        var remoteSink = exchangeService.newRemoteSink(groupTask, childSessionId, transportService, cluster.connection);
+                        exchangeSource.addRemoteSink(
+                            remoteSink,
+                            suppressRemoteFailure == false,
```

Comment:
The failFast param in addRemoteCluster may need to change to a Predicate Function that receives an `Exception` and returns a boolean. That way you can fail only if it is a remote-disconnected-exception (assuming that's the route we go with skip_unavailable).